src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java

```diff
@@ -112,19 +112,29 @@ public ProgrammingExerciseService(ProgrammingExerciseRepository programmingExerc
     }
 
     /**
-     * Notifies all particpations of the given programmingExercise about changes of the test cases.
+     * Notifies all participations of the given programmingExercise (including the template & solution participation!) about changes of the test cases.
+     * This method creates submissions for the participations so that the result when it comes in can be mapped to them.
      *
-     * @param programmingExercise The programmingExercise where the test cases got changed
+     * @param exerciseId of programming exercise the test cases got changed.
      */
-    public void notifyChangedTestCases(ProgrammingExercise programmingExercise, Object requestBody) {
-        for (StudentParticipation participation : programmingExercise.getParticipations()) {
-
-            ProgrammingExerciseStudentParticipation programmingExerciseStudentParticipation = (ProgrammingExerciseStudentParticipation) participation;
+    @Transactional
```

Comment:
I think you could leave this out, because the whole service is transactional, but it does not really matter

Reply:
You are right, I removed it!